tasks/Task.hpp

```diff
@@ -108,8 +114,47 @@ namespace usbl_evologics {
          * before calling start() again.
          */
         void cleanupHook();
-        void storePermanently();
-        void writeOutPosition();
+
+        /** Process I/O data
+         *
+         */
+        void processIO();
+
+        /** Get communication parameters
+         *
+         *  @return AcousticChannel with performance.
+         */
+        AcousticChannel getAcousticChannelparameters(void);
+
+        /** Update parameters on device.
+         *
+         * Compare actual with desired settings before update.
+         * @param desired_setting, parameters that should be applied on device.
+         * @param actual_setting, parameters present in device that will be used for compare.
+         */
+        void updateDeviceParameters(DeviceSettings const &desired_setting, DeviceSettings const &actual_setting);
+
+        /** Verify if free transmission buffer is big enough to support message.
```

Comment:
Can you give more information about that one ? You're dropping packages when it's not the case, but shouldn't you just wait to have enough space ? What's the implication ?